experiments/AtmosGCM/dcmip-3-1.jl

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init()
+using ClimateMachine.Atmos
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Thermodynamics: total_energy
+using ClimateMachine.VariableTemplates
+
+using Distributions: Uniform
+using LinearAlgebra
+using StaticArrays
+using Random: rand
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: R_d, day, grav, cp_d, cv_d, planet_radius, Omega, kappa_d
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+import CLIMAParameters
+CLIMAParameters.Planet.Omega(::EarthParameterSet) = 0.0
+CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 6.371 * 10^6 / 125
+
+
+function init_dcmip31!(bl, state, aux, coords, t)
+    FT = eltype(state)
+
+    ϕ = latitude(bl, aux)
+    λ = longitude(bl, aux)
+    z = altitude(bl, aux)
+
+    # initial velocity profile
+    u_0 = FT(20)
+    u = u_0 * cos(ϕ)
+    v = FT(0)
+    w = FT(0)
+
+    # surface temperature
+    _grav = FT(grav(bl.param_set))
+    _N = FT(0.01)
+    _cp = FT(cp_d(bl.param_set))
+    _Ω = FT(Omega(bl.param_set))
+    _a = FT(planet_radius(bl.param_set))
+    G = _grav^2 / (_N^2 * _cp)
+    T_eq = FT(300)
+    T_s = G  + (T_eq - G) * exp( -(u_0 * _N^2) / (4 * _grav^2) * (u_0 + 2 * _Ω *_a)* (cos(2ϕ) - 1))
+
+    # background temperature
+    T_b = G * (1 - exp((_N^2 / _grav)*z)) + T_s * exp((_N^2 / _grav)*z)
+
+    # surface pressure
+    p_eq = FT(100000)  # Pa
+    _R_d = FT(R_d(bl.param_set))
+    _kappa = FT(kappa_d(bl.param_set))
+    p_s = p_eq * exp( u_0 / (4 * G * _R_d) * (u_0 + 2 * _Ω *_a)* (cos(2ϕ) - 1)) * (T_s / T_eq)^(1/_kappa)
+
+    # unperturbed pressure field
+    p = p_s * (G / T_s * exp(-_N^2/_grav * z) + 1 - G/T_s)^(1/_kappa)
+
+    # Background potential temperature
+    θ_b = T_s * ( p_eq / p_s )^(_kappa) * exp(_N^2/_grav * z)
+
+    # density
+    ρ = p / (_R_d * T_b)
+
+    # potential temperature perturbation
+    Δθ = FT(1.0)
+    d = FT(5000)
+    λ_c = 2*FT(π)/3
+    ϕ_c = 0
+    r = _a * acos(sin(ϕ_c)*sin(ϕ) + cos(ϕ_c)*cos(ϕ)*cos(λ - λ_c))
+    s = d^2 / (d^2 + r^2)
+    L_z = FT(20000)
+    θ′ = Δθ * s * sin(2*FT(π)*z / L_z)
+
+    # Temperature perturbation
+    T′ = θ′ * (p / p_eq)^(_kappa)
+
+    e_pot = gravitational_potential(bl.orientation, aux)
+    e_kin = FT(0.5)*u^2
+
+    state.ρ = ρ
+    state.ρu = ρ * SVector{3, FT}(u, v, w)
```

Comment:
Have you verified that this preserves the HS equilibrium when the velocity is set to zero?